careercopilot/scripts/security-audit.sh

```
@@ -0,0 +1,438 @@
+    # Check for authentication middleware
+    if grep -qE "auth|jwt|token|security" backend/app/main.py; then
+        add_score 15 "Authentication middleware detected"
+    else
+        deduct_score 15 "Authentication middleware not detected"
+    fi
+fi
+
+# Check for environment variable usage
+if [ -f "frontend/.env" ]; then
+    # Check that API keys are not hardcoded
+    if find . -name "*.py" -o -name "*.js" -o -name "*.ts" -o -name "*.tsx" | xargs grep -l "AIzaSy\|sk-\|pplx-" 2>/dev/null | grep -v ".env"; then
```

Shellcheck (0.10.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)